sdk/go/common/resource/config/object.go

```diff
@@ -524,36 +524,10 @@ func isSecureValue(v any) (bool, string) {
 	return false, ""
 }
 
-func (c object) toDecryptedPropertyValue(decrypter Decrypter) resource.PropertyValue {
-	var prop resource.PropertyValue
-	switch v := c.value.(type) {
-	case bool, int64, float64, string:
-		if c.secure {
-			plaintext, err := decrypter.DecryptValue(context.Background(), v.(string))
-			contract.AssertNoErrorf(err, "failed to decrypt config")
```

Comment:
For the future: we cannot have asserts like this. It is reasonable for decryption to fail, and we should not crash if it does. We should return errors instead and let callers handle them.